Source/Core/VideoBackends/Metal/MTLRenderer.mm

```diff
@@ -0,0 +1,502 @@
+                                  float far_depth)
+{
+  g_state_tracker->SetViewport(x, y, width, height, near_depth, far_depth);
+}
+
+void Metal::Renderer::Draw(u32 base_vertex, u32 num_vertices)
+{
+  @autoreleasepool
+  {
+    g_state_tracker->Draw(base_vertex, num_vertices);
+  }
+}
+
+void Metal::Renderer::DrawIndexed(u32 base_index, u32 num_indices, u32 base_vertex)
+{
+  @autoreleasepool
+  {
+    g_state_tracker->DrawIndexed(base_index, num_indices, base_vertex);
+  }
+}
+
+void Metal::Renderer::DispatchComputeShader(const AbstractShader* shader,  //
+                                            u32 groupsize_x, u32 groupsize_y, u32 groupsize_z,
+                                            u32 groups_x, u32 groups_y, u32 groups_z)
+{
+  @autoreleasepool
+  {
+    g_state_tracker->SetPipeline(static_cast<const ComputePipeline*>(shader));
+    g_state_tracker->DispatchComputeShader(groupsize_x, groupsize_y, groupsize_z,  //
+                                           groups_x, groups_y, groups_z);
+  }
+}
+
+void Metal::Renderer::BindBackbuffer(const ClearColor& clear_color)
+{
+  @autoreleasepool
+  {
+    CheckForSurfaceChange();
+    CheckForSurfaceResize();
+    m_drawable = MRCRetain([m_layer nextDrawable]);
+    m_bb_texture->SetMTLTexture(MRCRetain([m_drawable texture]));
+    SetAndClearFramebuffer(m_backbuffer.get(), clear_color);
+  }
+}
+
+void Metal::Renderer::PresentBackbuffer()
+{
+  @autoreleasepool
+  {
+    g_state_tracker->EndRenderPass();
+    if (m_drawable)
+    {
+      [g_state_tracker->GetRenderCmdBuf()
+          addScheduledHandler:[drawable = std::move(m_drawable)](id) { [drawable present]; }];
+      m_bb_texture->SetMTLTexture(nullptr);
+      m_drawable = nullptr;
+    }
+    g_state_tracker->FlushEncoders();
+  }
+}
+
+std::unique_ptr<::BoundingBox> Metal::Renderer::CreateBoundingBox() const
+{
+  return std::make_unique<BoundingBox>();
+}
+
+void Metal::Renderer::CheckForSurfaceChange()
+{
+  if (!m_surface_changed.TestAndClear())
+    return;
+  m_layer = MRCRetain(static_cast<CAMetalLayer*>(m_new_surface_handle));
+  m_new_surface_handle = nullptr;
+  SetupSurface();
+}
+
+void Metal::Renderer::CheckForSurfaceResize()
+{
+  if (!m_surface_resized.TestAndClear())
+    return;
+  SetupSurface();
+}
+
+void Metal::Renderer::SetupSurface()
+{
+  CGSize size = [m_layer bounds].size;
+  // TODO: Update m_backbuffer_scale (need to make doing that not break everything)
+  float backbuffer_scale = [m_layer contentsScale];
+  size.width *= backbuffer_scale;
+  size.height *= backbuffer_scale;
+  [m_layer setDrawableSize:size];
+  m_backbuffer_width = size.width;
+  m_backbuffer_height = size.height;
+  TextureConfig cfg(m_backbuffer_width, m_backbuffer_height, 1, 1, 1, m_backbuffer_format,
+                    AbstractTextureFlag_RenderTarget);
+  m_bb_texture = std::make_unique<Texture>(nullptr, cfg);
+  m_backbuffer = std::make_unique<Framebuffer>(m_bb_texture.get(), nullptr,  //
```

Comment:
Stray `//`?

Reply:
Structural `//`.  Keeps width and height next to each other.